lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -74,6 +79,18 @@ def domain_id
     end
 
     private
+    def domain
```

Comment:
Now I do run into the problem where I need to mock the mysql query during initialization. Any hint on how to do that?